### PR TITLE
feat: MenuHeading and MenuDivider

### PR DESCRIPTION
**Pull request by ElliotTheHuman**
- Added `MenuHeading` + `MenuDivider` to replace `MenuGroup` existing behaviors
- Replaced existing storybook usage of `MenuGroup` with `MenuHeading` + `MenuDivider`

### 🔧  TODO (Follow-up PRs)
- Create docs for `MenuDivider` + `MenuHeading`
- Delete `MenuGroup`
- Update `Menu`-related docs to use `MenuDivider` + `MenuHeading`

### Requirements

Please check the following items are addressed in your pull request (or are not applicable)

- [ ] a11y impact (FUTURE: Make aXe pass req'd for CI ✅)

#### Image snapshots (choose one)

  - [ ] yes
  - [ ] not applicable

#### Documentation updated (choose one)

  - [ ] yes
  - [ ] not applicable

### CODE REVIEW

File: packages/components/src/Menu/MenuHeading.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import { useElementVisibility } from './MenuHeading.hooks'
+
+export const MenuHeading: FC = ({ children }) => {
+  const labelShimRef: RefObject<any> = useRef()
+  const isLabelShimVisible = useElementVisibility(labelShimRef)
+
+  const { density } = useContext(ListItemContext)
+  const { height, labelFontSize, labelLineHeight, px, py } = listItemDimensions(
+    density
+  )
```
Comment (ElliotTheHuman) [inline packages/components/src/Menu/MenuHeading.tsx:42]: Aside from the fact that `MenuHeading` respects parent `Menu` / `MenuList`'s "density" value, this file is a copy of `MenuGroupLabel`

File: packages/components/src/Menu/MenuHeading.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      <div ref={labelShimRef} style={{ height: '0' }} />
+      <Heading
+        as="h2"
+        color="text5"
+        fontFamily="body"
+        fontSize="xsmall"
+        fontWeight="medium"
+        lineHeight="xsmall"
+        px={px}
+        py="xsmall"
```
Comment (ElliotTheHuman) [inline packages/components/src/Menu/MenuHeading.tsx:59]: These specs were from the design maestro, Jared!
https://www.figma.com/file/zd6MjUXIIstvG7gTJWaX2Y/Looker-Components?node-id=11194%3A93767

**Review by mdodgelooker — CHANGES_REQUESTED** [review]
Very cool! Had some suggestions on `MenuDivider` styling, and I think codecov is complaining.

File: packages/components/src/Menu/MenuDivider.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+
+import styled from 'styled-components'
+
+export const MenuDivider = styled.li`
+  border-top: 1px solid ${({ theme: { colors } }) => colors.ui2};
+`
```
[the remark this answers is not in the document]
Comment (ElliotTheHuman) [reply packages/components/src/Menu/MenuDivider.tsx:31]: The & + & is awesome! Thanks for the suggestion :) 

Regarding divider's spacing, I took another look at [Jared's design specs](https://www.figma.com/file/zd6MjUXIIstvG7gTJWaX2Y/Looker-Components?node-id=1861%3A0) and it looks like it's calling for 8px margin above and below the divider, in which case I can push that change up.

That being said, I'm cool with using density to determine spacing as well though if we like that approach better!

@jhardy Thoughts on whether a constant margin-y or using density for margin-y is a better design call here?

[the remark this answers is not in the document]
Comment (ElliotTheHuman) [reply packages/components/src/Menu/MenuDivider.tsx:31]: (Wanted to note that we're not scaling `MenuHeader` with density, so wondering if that tips the scale in favor of not scaling `MenuDivider` as well 🤔 )

File: packages/components/src/Menu/Menu.story.tsx [hunk 4]
```diff
@@ -448,19 +398,6 @@ export const LongMenus = () => {
...
-              {items.map((item, index2) => (
-                <MenuItem key={`${item.label}-${index2}`}>
-                  {item.label}
-                </MenuItem>
-              ))}
-            </MenuGroup>
-          ))}
-        >
-          <Button>No windowing (groups)</Button>
-        </Menu>
```
[the remark this answers is not in the document]
Comment (ElliotTheHuman) [reply packages/components/src/Menu/Menu.story.tsx:463]: I didn't even know Menu had sticky headers. They look real slick 😍 

My fault for removing this in the first place.

File: packages/components/src/Menu/MenuList.story.tsx [hunk 5]
```diff
@@ -75,7 +78,8 @@ Basic.args = {
           {item.children}
         </MenuItem>
       ))}
-    </MenuGroup>
+      {key < groupsArray.length - 1 && <MenuDivider />}
```
[the remark this answers is not in the document]
Comment (ElliotTheHuman) [reply packages/components/src/Menu/MenuList.story.tsx]: Resolved in 4dbdb2b

File: packages/components/src/Menu/Menu.story.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (ElliotTheHuman) [reply packages/components/src/Menu/Menu.story.tsx]: Added back in 4dbdb2b

**Comment by ElliotTheHuman** [comment]
First round of feedback implemented! (Note: left some thoughts regarding `MenuDivider` margin-y. TLDR is that Jared's design specs call for 8px constant across densities, but I pinged him to see if it's better to have it scale)

File: packages/components/src/Menu/MenuDivider.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (ElliotTheHuman) [reply packages/components/src/Menu/MenuDivider.tsx]: Spoke to Jared, and he suggested we start simple for now and have 8px for margin-y across the board.

[the remark this answers is not in the document]
Comment (mdodgelooker) [reply packages/components/src/Menu/MenuDivider.tsx]: Sounds good! Now I'm just trying to remember what we do about new styled components that bring coverage down... Just ignore? Smoke test by rendering once in Jest?
image

[the remark this answers is not in the document]
Comment (mdodgelooker) [reply packages/components/src/Menu/MenuDivider.tsx]: Ah sounds like a jest smoke test is in order. Maybe just add a super simple render test with `MenuDivider` and `MenuHeading` in `MenuList.test.tsx`?

[the remark this answers is not in the document]
Comment (ElliotTheHuman) [reply packages/components/src/Menu/MenuDivider.tsx]: Good call. Done!

62d32dd

**Review by mdodgelooker — APPROVED** [review]
LGTM!

**Pull request merged**